### PR TITLE
Don't use closeTimer in sendbuffer

### PR DESCRIPTION
**Pull request by oxtoacart**
@joesis After I merged your PR, I realized that we're reading from closeTimer on a different goroutine than we're resetting it. This PR gets rid of closeTimer and just uses closing a channel to achieve the same thing.

### CODE REVIEW

File: sendbuffer.go [hunk 1]
```diff
@@ -60,16 +60,19 @@ func (buf *sendBuffer) sendLoop(w io.Writer) {
...
-		buf.muClosing.Unlock()
-		if !closeTimer.Stop() {
-			<-closeTimer.C
-		}
-		closeTimer.Reset(closeTimeout)
+		go func() {
+			buf.muClosing.Lock()
+			alreadyClosed := buf.closing
+			buf.closing = true
+			close(buf.in)
```
Comment (joesis) [inline sendbuffer.go]: If `alreadyClosed` can be true, does it mean `buf.in` can be closed more than once, which could cause a panic?

Comment (joesis) [reply sendbuffer.go]: `sync.Once` seems a good fit

Comment (oxtoacart) [reply sendbuffer.go]: Good idea, done!

**Comment by joesis** [comment]
LGTM. Thanks for the catch and fix @oxtoacart !

**Comment by oxtoacart** [comment]
Thanks @joesis!

**Pull request merged**